src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.cs

```diff
@@ -3,6 +3,7 @@
 // See the LICENSE file in the project root for more information.
```

Comment:
I made thsi change because debugging thsi code was a nightmare.  THings were lazy and computed *numerous* time as linq expressions were called all over the place.  That's just wasteful, and we need all the values anyways.  So we just realize everything and only do these computations once.  It's much simpler to understand/debug now.